### PR TITLE
remove stacktrace.ContextSetter

### LINKED ISSUE #646: Remove stacktrace.ContextSetter
It's currently possible to set stacktrace "context" — surrounding lines of source code — by supplying a stacktrace.ContextSetter to Tracer. This is almost certainly not being used, and will become irrelevant once we have integration with Code Search. We should remove the ContextSetter bits from the agent in a future major version.

### PR DESCRIPTION
**Pull request by stuartnelson3**
closes #646

### CODE REVIEW

File: modelwriter.go [hunk 1]
```diff
@@ -262,19 +259,6 @@ func stacktraceCulprit(frames []model.StacktraceFrame) string {
...
-func (w *modelWriter) setStacktraceContext(stack []model.StacktraceFrame) {
-	if w.cfg.contextSetter == nil || len(stack) == 0 {
-		return
-	}
-	err := stacktrace.SetContext(w.cfg.contextSetter, stack, w.cfg.preContext, w.cfg.postContext)
-	if err != nil {
-		if w.cfg.logger != nil {
-			w.cfg.logger.Debugf("setting context failed: %v", err)
-		}
-		w.stats.Errors.SetContext++
```
Comment (axw) [inline modelwriter.go:274]: can we get rid of the SetContext field now?

File: tracer.go [hunk 2]
```diff
@@ -525,7 +524,6 @@ type tracerConfig struct {
 	metricsInterval         time.Duration
 	logger                  WarningLogger
 	metricsGatherers        []MetricsGatherer
-	contextSetter           stacktrace.ContextSetter
 	preContext, postContext int
```
Comment (axw) [inline tracer.go:527]: Do we still need preContext/postContext? I think we can get rid of all related config and env vars?

**Review by axw — COMMENTED** [review]
Thanks, looks good. I think there might be a few more related things we can get rid of.

File: tracer.go (hunk 2 above)
Comment (stuartnelson3) [reply tracer.go]: seems like it, removed :magic_wand:

**Review by axw — APPROVED** [review]
:tada: LGTM!

**Comment by stuartnelson3** [comment]
I didn't see any additional environment variables, but I cleaned up the pre/post context stuff, and code related to Errors.SetContext

**Comment by simitt** [comment]
@axw anything you think we should test for this? Nothing comes to my mind, so I'd skip manual testing on this one.

**Comment by axw** [comment]
Nope, nothing to test here, we just took away an API.

**Pull request merged**